### PR TITLE
Update opentelemetry-mapping-go to v0.9.0

### PR DESCRIPTION
**Pull request by mx-psi**
### What does this PR do?

Bumps opentelemetry-mapping-go to v0.9.0.

### Motivation

Include:

- DataDog/opentelemetry-mapping-go/pull/218 and DataDog/opentelemetry-mapping-go/pull/220
- DataDog/opentelemetry-mapping-go/pull/219
- DataDog/opentelemetry-mapping-go/pull/213
- DataDog/opentelemetry-mapping-go/pull/202

### Additional Notes

The metrics change does not affect the OTLP ingest pipeline, since the Agent does not have the internal metrics pipeline enabled

### Describe how to test/QA your changes

Send an OTLP log to the Agent and verify that milliseconds are shown in the timestamp on the UI.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

[CONTINT-258]: https://datadoghq.atlassian.net/browse/CONTINT-258?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by mx-psi** [comment]
Blocked by #21501

**Review by mackjmr — APPROVED** [review]

File: releasenotes/notes/otel-mapping-v0.9.0-a1a7a3f8d501cf6f.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+enhancements:
+  - |
+    OTLP ingest log timestamps (i.e. '@timestamp') now include milliseconds
```
Comment (drichards-87) [inline releasenotes/notes/otel-mapping-v0.9.0-a1a7a3f8d501cf6f.yaml:11]:
```suggestion
    OTLP ingest log timestamps (i.e. '@timestamp') now include milliseconds.
```

**Review by drichards-87 — APPROVED** [review]
Left you the tiniest of suggestions on the release note.

**Review by vickenty — APPROVED** [review]

**Comment by mx-psi** [comment]
@DataDog/agent-apm any chance you can review this? :pray:

**Review by ajgajg1134 — APPROVED** [review]
looks good from agent apm!

**Pull request merged**